### ISSUE TITLE
Tv gaat telkens op mute na installatie van de app

### ISSUE THREAD

**Issue by AVDP86** [RonnyWinkler/homey.sonybraviaandroidtv#2]
Ik ben super blij met de app voor de sony tv. En doet ook wat hij moet doen. Alleen na een poosje stuurt deze regel matig een signaal voor mute naar de tv.

**Comment by RonnyWinkler** [comment]
Please update your post with an english version. Thanks.

**Comment by AVDP86** [comment]
I’m sorry, i’m verry pleased with the app. It does what it needs to but after installing the app it seems to send radomly signals to my tv en turns it on mute.

**Comment by RonnyWinkler** [comment]
Sounds strange. The app only sends commands on a button press or using flow actions. Are you sure you don't have a flow that triggers this setting?
And oll other actions are working as espected?

**Comment by AVDP86** [comment]
Yes everything works as it should. Ik know it sounds strange. I had a few sony apps running.
Deleted them one by one. And the problem stopped. Now yesterday installed it again and so it began again. There’s nothing in my timeline and i deleted all possible flows involving the tv

**Comment by oleast** [comment]
I can confirm that I'm experiencing the same problem! Irritated me for months, until I thought it could come from Homey. Resolved immediately after uninstalling this app.

My TV is a KD55A85, running latest available software.

I could try to help debug if you want?

**Comment by RonnyWinkler** [comment]
@oleast 
Can you describe for example:
- how or when does it happen?
- periodically or random?
- was the TV muted before and then unmuted with the remote?
- are you using a kind of widget or are you using the Homey device outside Homey (HA vie Mqtt or similar)?
- Is the app restarted periodically? Or Homey itself?

The app itself won't send commands on it own. But perhaps there is something changing device the capabilities.

**Comment by oleast** [comment]
I'll reinstall the app and try to find some correlations over the next few days. In the meantime this is what I can recall:

- It seems to happen completly at random. Sometimes it can happen with many times just minutes apart, other times be hours between events. But it would happen almost every single day while using the TV.
- There seems to be no correlation between this happening and the TV being muted before. I could shut the TV completly off by removing the power cable, turn it on again, and a mute event happening 5 minutes after I turned the TV on. When it happens I would typically unmute it with the remote straight away. I've never experienced that it would unmute after it has been muted, but it is hard to say if this is because I've always unmuted it manually before a new event happened.
- I have Homey connected to both Google Home and Apple Homekit, but I'm not connected to anything else that is external that I can see.
- I'll have a look at potential restarts of the app. I would not expect that it comes from Homey itself restarting, since the muting could happen very frequently.

I had this problem for months before finding out it even came from Homey or this app at all. At first I thought it came from an error in my sound system.

**Comment by oleast** [comment]
@RonnyWinkler 
I set up the app this weekend, but got my first "mute" action just now. If the diagnostic report can help you can have a look at it: `83d5d522-e59b-4b37-93d0-ce3f89f97245`

I also set up a few actions that trigger "on tv muted" in Homey to help debug.
- First I set up a push notification to my phone that tells me the TV was muted.
- Then I set up a flow that unmutes the TV whenever it is muted.
- I also have the polling interval set to 30 seconds.

This means that is I mute the TV manually with the remote, then it will automatically unmute when Homey polls the TV.

When the automatic muting event happened, I did not notice the TV actually muting, I just noticed the push notification. I think this tells us that there was almost no time between the automatic muting and my homey flow unmuting again.
Which means that either the muting is happening during the polling loop to the TV, or I was just really lucky and the automatic muting happened at the same time as the polling interval.

**Comment by RonnyWinkler** [comment]
Are you using headphones in this case (or earlier before turning the TV off)?
I have the suspicion that the TV reports speaker and headphone muting state in this case and the app takes the wrong.

**Comment by oleast** [comment]
I am not using headphones, but I am using a soundbar connected over HDMI (eARC). It is constantly connected, and random mute events can happen when I have not changed to volume since turing on the TV.

**Comment by RonnyWinkler** [comment]
@oleast 
I think the issue happens if the TV reports several mute states (speaker + soundbar). Then perhaps the wrong entry is used. I can't see it in the logs (that's not logged in detail). But bases on the code it's a possible reason. I fixed the code that only the speaker state us used now.
Can you please update to new test version and check the next days? Thanks.
https://homey.app/a/name.ricardoismy.sonybraviaandroidtv/test/

**Comment by AVDP86** [comment]
- @oleast thanks for joining in. @RonnyWinkler Just two weeks ago, i'fe started having other problems with my tv that have drove me to reset to factory settings. Until now i haven't yet installed the app again. So i wil try the test version. I also had homekit connected. My idea is to not connect to homekit. Only homey and this app. Thanks for the help!

**Comment by RonnyWinkler** [comment]
> * started having other problems with my tv

After AndroidTV update? I also have problems with the TV app that starts to hang after some channel changes. Factory reset didn't work.
